cmd/bootstrapper/generator.go

```diff
@@ -56,23 +57,42 @@ func WithFilesystem(fs afero.Fs) Opt {
 	}
 }
 
-func WithHttpClient(c *http.Client) Opt {
+func WithHttpClient(c *retryablehttp.Client) Opt {
 	return func(g *Generator) {
 		g.client = c
 	}
 }
 
+func checkRetry(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	if resp != nil && resp.StatusCode != http.StatusOK {
+		return true, nil
+	}
+	return retryablehttp.DefaultRetryPolicy(ctx, resp, err)
+}
+
 func NewGenerator(btcEndpoint, smEndpoint string, opts ...Opt) *Generator {
 	g := &Generator{
 		logger:      zap.NewNop(),
 		fs:          afero.NewOsFs(),
-		client:      &http.Client{},
 		btcEndpoint: btcEndpoint,
 		smEndpoint:  smEndpoint,
 	}
+
 	for _, opt := range opts {
 		opt(g)
 	}
+
+	if g.client == nil {
+		client := &retryablehttp.Client{
+			RetryMax:     10,
+			RetryWaitMin: 5,
+			RetryWaitMax: 2 * 5,
```

Comment:
I believe those are nano seconds if you don't specify them:

```suggestion
			RetryWaitMin: 5 * time.Second
			RetryWaitMax: 10 * time.Second,
```